### PR TITLE
Adds Lean Recipe 

### PR DESCRIPTION
**Pull request by RiskySikh**
## About The Pull Request

image

Greetings friends, this PR adds a craftable recipe to make the drink called Lean. You might be asking what is lean? Well, lean was originally created in Houston, Texas in the 1990s during the hip-hop scene. It was made with sprite, jolly ranchers, and codeine. There has been alterations in this recipe instead of sprite we have space-up, instead of jolly ranchers you got gumballs, instead of codeine you got morphine which codeine is derived from. It is simple to make, and easily accessible. To get colocup, you get plastic from the autolathe and turn it into a colocup, space-up you get it from the vending machine, gumballs from your local mediborg, and morphine from medical. I also forgot you need 1 colocup, 2 gumballs, 15 units of space-up, and 1 unit of morphine.  The effects from drinking lean is slurred speech, drugged, and drowsiness. 

## Why It's Good For The Game

Well, more recipes are beneficial to the game, and it's gives an opportunity to assistants and  other people who are bored with their day to day lives. This drink is mainly focused on assistants since they can craft this easily available drink and drink away their sorrows of being the most discrimination, most marginalized people on the station. The recipe is basic, and is not complex; however, it provides some sort of a journey to gather all of the ingredients, and the reward is a nice refreshing drink of lean.

## Changelog
:cl:
add: Added a new lean recipe which can be accessed in the crafting menu under drinks 
/:cl:

### CODE REVIEW

**Comment by yeeyeh** [comment]
nice essay, thank you bindy johal, proud of you

**Comment by wesoda25** [comment]
one small step for assistant, one giant step for trap culture

**Comment by Shadowflame909** [comment]
Ian Gang (ft Security Officer)

**Comment by Twaticus** [comment]
SKRR SKRR

**Comment by RiskySikh** [comment]
Credits to Kryson who made these wonderful sprites

**Comment by cacogen** [comment]
>the guy who plays bindy is one of the few people who put in the effort to properly describe what his pull requests do

**Comment by necromanceranne** [comment]
Redemption arc is strong with this one.

**Comment by MrDoomBringer** [comment]
[thread theme video](https://lew.la/reddit/clips/e9k9c0.mp4)

**Comment by evilcyborgmrt** [comment]
Traditionally made using codeine syrup rather than crushed up pills, linguists continue to debate whether the name lean is a portmanteau of liquid and codeine; or whether it refers to the leaning position induced by its consumption

**Comment by dablusniper** [comment]
I think we should have some alternative way to obtain gumballs so the recipe isn't dependent on whether a medborg is present or not. Maybe add them to the Getmore vending machines?

**Comment by Krysonism** [comment]
I think you should increase the morphine requirement to 10u, or at least 5u. 

A 30u bottle is quite cheap from the vendor and 1-2 bottles can usually be stolen from medbay for free.

**Comment by RiskySikh** [comment]
> 
> 
> I think we should have some alternative way to obtain gumballs so the recipe isn't dependent on whether a medborg is present or not. Maybe add them to the Getmore vending machines?
> 
> Otherwise, pretty epic.

I know that Inept is making a gumball machine PR.

**Comment by RiskySikh** [comment]
> 
> 
> I think you should increase the morphine requirement to 10u, or at least 5u.
> 
> A 30u bottle is quite cheap from the vendor and 1-2 bottles can usually be stolen from medbay for free.

I was thinking about it, but I was afraid of accidentally overdosing the person with morphine.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -914,3 +914,25 @@
 		M.adjustToxLoss(-1, 0)
 	..()
 	. = TRUE
+
+/datum/reagent/consumable/lean
+	name = "Lean"
+	description = "The drank that makes you go wheezy."
+	color = "#DE55ED"
+	quality = DRINK_NICE
+	taste_description = "you can taste purple and a hint of opiod."
```
Comment (nemvar) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Taste description doesn't work this way. The message gets constructed of all reagents that are contained.
```suggestion
	taste_description ="purple with a hint of opoid"
```

**Comment by RiskySikh** [comment]
Gotcha nemvar, fixed it

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -914,3 +914,24 @@
...
+	glass_desc = "A drink that makes your life less miserable. "
+
+/datum/reagent/consumable/lean/on_mob_life(mob/living/carbon/M)
+	if(M.slurring < 3)
+		M.slurring+= 2
+	if(M.druggy < 3)
+		M.adjust_drugginess(1)
+	if(M.drowsyness < 3)
+		M.drowsyness++
+	. = ..()
```
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	return ..()
```

**Comment by ExcessiveUseOfCobblestone** [comment]
how are you ODing on something that isn't the catalyst or result in the reaction?

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 3]
```diff
@@ -513,12 +513,14 @@
 	volume = 20
 	amount_per_transfer_from_this = 5
 	isGlass = FALSE
+	var/random_sprite = TRUE
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/food_and_drinks/drinks/drinks.dm]: doc

Comment (RiskySikh) [reply code/modules/food_and_drinks/drinks/drinks.dm]: > 
> 
> doc

Alright added the proper documentation for var/random_sprite = TRUE

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 4]
```diff
@@ -513,6 +513,11 @@
 	volume = 20
 	amount_per_transfer_from_this = 5
 	isGlass = FALSE
+	/** 
+	  * The var fixes the lean sprite to work properly upon crafting. Before crafting, it will 
+	  * display the random sprite; however after crafting it the random sprite becomes false 
+	  * and displays the lean sprite
+	  */
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/food_and_drinks/drinks/drinks.dm]: bruh https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-variable

Comment (RiskySikh) [reply code/modules/food_and_drinks/drinks/drinks.dm]: > 
> 
> bruh https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-variable

there you I commented it properly and made it 100% direct to the person who reads it

**Comment by ExcessiveUseOfCobblestone** [comment]
you broke it

**Comment by RiskySikh** [comment]
> 
> 
> you broke it

it's fixed

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 5]
```diff
@@ -913,3 +913,25 @@
...
+
+/datum/reagent/consumable/lean
+	name = "Lean"
+	description = "The drank that makes you go wheezy."
+	color = "#DE55ED"
+	quality = DRINK_NICE
+	taste_description = "purple and a hint of opiod."
+	glass_icon_state = "lean"
+	glass_name = "Lean"
+	glass_desc = "A drink that makes your life less miserable. "
```
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	glass_desc = "A drink that makes your life less miserable."
```

**Comment by bruhlookatthisdood** [comment]
lean
gone

**Pull request merged**